### PR TITLE
Diemo

### PR DESCRIPTION
**Pull request by pakb**
**NOTE: this branch is based on branch `layerdefs` DO NOT merge any branch into this one!! If you need upstream changes in master: rebase (not merge, rebase!) layerdefs onto master, then diemo onto layerdefs!!**

replaces https://github.com/geoadmin/mf-chsdi3/pull/3155 (change in git branch name as `]` char might not be a good idea for URLs)

### CODE REVIEW

File: production.ini.in [hunk 1]
```diff
@@ -21,7 +21,8 @@ pyramid.includes = pyramid_tm
 sqlalchemy.are.url = postgresql://${dbhost}:${dbport}/are_${dbstaging}
 sqlalchemy.bafu.url = postgresql://${dbhost}:${dbport}/bafu_${dbstaging}
 sqlalchemy.bak.url = postgresql://${dbhost}:${dbport}/bak_${dbstaging}
-sqlalchemy.bod.url = postgresql://${dbhost}:${dbport}/bod_${dbstaging}
+# TODO: once we want this in production, change back and change the links in bod master layer_js table.
+sqlalchemy.bod.url = postgresql://${dbhost}:${dbport}/bod_ltkum_master
```
Comment (pakb) [inline production.ini.in]: do not forget to switch back to `bod_master` before going live

**Comment by pakb** [comment]
test link on INT here : https://mf-chsdi3.int.bgdi.ch/diemo/shorten/82d8d2de5a

File: production.ini.in (hunk 1 above)
Comment (ltkum) [reply production.ini.in]: I'll do my best

File: chsdi/static/vectorStyles/ch.bfe.ladestellen-elektromobilitaet.json [hunk 2]
```diff
@@ -1,68 +1,261 @@
...
-	"property": "Available",
+	"property": "Availability",
 	"values": [{
+                "geomType": "point",
+                "value": "Unknown",
+                "minResolution":0,
+                "maxResolution":20,
+                "vectorOptions": {
+                        "type": "icon",
+                "src": "https://data.int.bgdi.ch/ch.bfe.ladestellen-elektromobilitaet/symbols/icon-pin-shadow-no-status-information-50px.png",
```
Comment (AFoletti) [inline chsdi/static/vectorStyles/ch.bfe.ladestellen-elektromobilitaet.json]: A question for @ltkum 
Is there a reason to have the icons on data.int and not data.geo?
If not, just tell and I will take care of this small fix

**Comment by AFoletti** [comment]
We absolutely have to check if [the DIEMO chsdi model](https://github.com/geoadmin/mf-chsdi3/pull/3223/files#diff-866ae80c49bcc1a4786d714e1233d625) works with the new DB schema.
Moreover, we need to expand it to include the various new attributes.
@danduk82 something for @rebert ?

**Comment by rebert** [comment]
@danduk82 FYI
@AFoletti  the model is pointing to a view and the view has not changed, the new attributes have not been added. So CHSDI shoud still work, as the view did not change. For me this is a feature request ... I would mind, for the next time, to do this earlier in the project....? Do you want to add all new attributes to the view/to the model? or which attributes are you missing? Does it make sense to add them all? Does it make sense to add them? What is the use case?

Tobias

**Comment by AFoletti** [comment]
@rebert 
I asked the customer directly in another ticket for confirmation, but I guess that having the CHSDI output to deliver all of the OICP attributes (with the addition of your special "Query" ones) is a safe bet.
But let's not make the job twice needlessly.
I think we can wait for an official answer (which will come next week probably) and then adapt. Would a day of work be enough to add all the new attributes to the view and the chsdi model?

**Comment by rebert** [comment]
@AFoletti good idea. Yes, a day should absolutely be enough.

**Comment by AFoletti** [comment]
@boecklic I changed the icon and database sources. This PR is in my opinion ready to be merged as preparation for deploy next week.
I will still commit and merge the last version of the diemo legends (still pending delivery) but that's a minor thing and can be done via a new branch

**Pull request merged**